src/test/incomparable.rs

```diff
@@ -0,0 +1,88 @@
+use quote::quote;
+use syn::Result;
+
+use super::test_derive;
+
+#[test]
+fn enum_() -> Result<()> {
+	test_derive(
+		quote! {
+			#[derive_where(PartialEq, PartialOrd)]
+			enum Test{
+				A,
+				#[derive_where(incomparable)]
+				B(Box<dyn Fn()>),
+				C(String),
+				#[derive_where(incomparable)]
+				D,
+				#[derive_where(incomparable)]
+				E { test: u8 }
+			}
+		},
+		quote! {
+			impl ::core::cmp::PartialEq for Test {
+				#[inline]
+				fn eq(&self, __other: &Self) -> bool {
+					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
+						match (self, __other) {
+							(Test::C(ref __0), Test::C(ref __other_0)) => true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::B(..) | Test::D | Test::E {..}, ..) => false,
+							_ => true,
+						}
+					} else {
+						false
+					}
+				}
+			}
+			impl ::core::cmp::PartialOrd for Test {
+				#[inline]
+				fn partial_cmp(&self, __other: &Self) -> ::core::option::Option<::core::cmp::Ordering> {
+					if ::core::matches!(self, Test::B(..) | Test::D | Test::E { .. })
+						|| ::core::matches!(__other, Test::B(..) | Test::D | Test::E { .. })
+					{
+						return ::core::option::Option::None;
+					}
+					let __self_disc = ::core::mem::discriminant(self);
+					let __other_disc = ::core::mem::discriminant(__other);
+					if __self_disc == __other_disc {
+						match (self, __other) {
+							(Test::C(ref __0), Test::C(ref __other_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+									::core::option::Option::Some(::core::cmp::Ordering::Equal) =>
+										::core::option::Option::Some(::core::cmp::Ordering::Equal),
+									__cmp => __cmp,
+								},
+							_ => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
+						}
+					} else {
+						::core::cmp::PartialOrd::partial_cmp(
+							&unsafe { ::core::mem::transmute::<_, isize>(__self_disc) },
+							&unsafe { ::core::mem::transmute::<_, isize>(__other_disc) },
+						)
+					}
+				}
+			}
```

Comment:
@daxpedda The question here is, does it make sense to test all the different outputs? i.e. safe/unsafe/nightly or just one of them and disable the test in all other configurations?

Reply:
We are already testing all configurations in other tests, we don't have to test them all over again for every single option.

Instead of skipping we could just test with `contains(...)` or something like that, but we can also skip, I don't mind.